perfkitbenchmarker/sample.py

```diff
@@ -33,10 +34,12 @@ def PercentileCalculator(numbers):
   count = len(numbers_sorted)
   total = sum(numbers_sorted)
   result = {}
-  for percentile in PERCENTILES_LIST:
+  for percentile in percentiles:
     percentile_string = 'p%s' % str(percentile)
-    result[percentile_string] = numbers_sorted[
-        int(count * float(percentile) / 100)]
+    index = int(count * float(percentile) / 100.0)
+    if index == count:
```

Comment:
If the "percentiles" passed in is something like "101.0", then it would still overflow. (the original code works because it works on a constant list of percentiles up to 99.9), now it takes in an input list from the user, we might want to do some validation?
also, you can probably simplify the if check as one liner: index = min(index, count - 1)